Initialize all of the fields of the key to zeroes, so that we don't pass
garbage keys in to later versions of Berkeley DB.
--- nss_db-2.2/db-XXX.c	2005-03-29 16:13:57.103516704 -0500
+++ nss_db-2.2/db-XXX.c	2005-03-29 16:13:47.108821931 -0500
@@ -225,6 +225,7 @@
   DBT key;								      \
   enum nss_status status;						      \
   const size_t size = (keysize) + 1;					      \
+  memset(&key, 0, sizeof(key));						      \
   key.data = alloca (size);						      \
   key.size = KEYPRINTF keypattern;					      \
   key.flags = 0;							      \
